lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/DriverStatusHTTPHandler.java

```diff
@@ -126,4 +164,29 @@ String waitAndGetMessage() {
   static String getMessageForStatus(final ReefServiceProtos.JobStatusProto status) {
     return status.getState().name();
   }
+
+  /**
+   * Schedules an alarm, if needed.
+   * <p>
+   * The alarm will prevent the Clock from going idle. This gives the .NET Client time to make a call to this HTTP
+   * handler.
+   */
+  private void scheduleAlarm() {
+    if (wasCalledViaHTTP || retry >= maxNumberOfRetries) {
+      // No alarm necessary anymore.
+      LOG.log(Level.INFO,
+          "Not scheduling additional alarms after {0} out of max {1} retries. The HTTP handles was called: ",
+          new Object[]{retry, maxNumberOfRetries, wasCalledViaHTTP});
+      return;
+    }
+
+    // Scheduling an alarm will prevent the clock from going idle.
+    ++retry;
+    clock.scheduleAlarm(alarmInterval, new EventHandler<Alarm>() {
```

Comment:
actually, we can create one instance of `EventHandler<Alarm>()` in a member variable, i.e.
```java
private final EventHandler<Alarm> alarmHandler = new EventHandler<Alarm>() {
  @Override
  public void onNext(final Alarm value) {
    scheduleAlarm();
  }
};
```
and then
```java
clock.scheduleAlarm(alarmInterval, alarmHandler); 
```
so we don't have to create a new instance on each retry

Reply:
Cool! Done.